automation/src/test/scala/org/broadinstitute/dsde/test/api/AuthDomainMultiGroupApiSpec.scala

```diff
@@ -0,0 +1,418 @@
+                    AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+        }
+
+        "when the user is a billing project owner" - {
+          "can be seen but is not accessible" in {
+
+            val user = UserPool.chooseProjectOwner
+            implicit val authToken: AuthToken = authTokenDefault
+
+            withGroup("AuthDomainOne") { groupOne =>
+              withGroup("AuthDomainTwo") { groupTwo =>
+
+                withCleanBillingProject(user, userEmails = List(defaultUser.email)) { projectName =>
+                  withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo)) { workspaceName =>
+
+                    // user can see workspace but user cannot access workspace
+                    AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+        }
+
+        "when not shared with them" - {
+          "cannot be seen and is not accessible" in {
+            val user = UserPool.chooseStudent
+            implicit val authToken: AuthToken = authTokenDefault
+
+            withGroup("AuthDomainOne") { groupOne =>
+              withGroup("AuthDomainTwo") { groupTwo =>
+
+                withCleanBillingProject(defaultUser) { projectName =>
+                  withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo)) { workspaceName =>
+
+                    // user cannot see workspace and user cannot access workspace
+                    AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+
+      // one group
+      "when the user is in one of the groups" - {
+        "when shared with them" - {
+          "can be seen but is not accessible" in {
+
+            val user = UserPool.chooseStudent
+            implicit val authToken: AuthToken = authTokenDefault
+
+            withGroup("AuthDomainOne") { groupOne =>
+              withGroup("AuthDomainTwo", List(user.email)) { groupTwo =>
+
+                withCleanBillingProject(defaultUser) { projectName =>
+
+                  withWorkspace(projectName, "GroupsApiSpec_workspace", Set(groupOne, groupTwo), List(AclEntry(user.email, WorkspaceAccessLevel.Reader))) { workspaceName =>
+
+                    // user can see workspace but user cannot access workspace
+                    AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+                  }
+                }
+              }
+            }
+          }
+
+          "when the user is a billing project owner" - {
+            "can be seen but is not accessible" in {
```

Comment:
this test should include two auth domains, and `user` should be in one of them

Reply:
👍